spec/forms/totp_verification_form_spec.rb

```diff
@@ -33,5 +33,23 @@
         expect(form.submit).to eq result
       end
     end
+
+    context 'when the code is not exactly Devise.otp_length digits' do
+      it 'returns FormResponse with success: false' do
+        user = build_stubbed(:user)
+        codes = %W(123abc 1234567 abcdef aaaaa\n123456\naaaaaaaaa)
```

Comment:
Yes that makes sense. I was looking for more explicitness here around why these are not valid. If we label them as such, that would be great.